### PR TITLE
Fix configuration not saved after migration to new service management

### PR DESCRIPTION
**Pull request by kernevil**
Signed-off-by: Samuel Cabrero <scabrero@suse.de>

### CODE REVIEW

**Comment by kernevil** [comment]
@dgdavid Could you please have a look? I am not sure why unit test is failing, running it locally in Tumbleweed passes.

**Comment by dgdavid** [comment]
Thank you @kernevil for submitting the fix.

I think that tests are failing because now it is necessary to mockup the `SambaServer.WriteDialog` properly for both cases: when the configuration is written successfully and when it is not.

In fact, I just clone your repo and perform the below changes in the `fix_config_save` branch and tests are passing again.

```diff
diff --git a/src/include/samba-server/complex.rb b/src/include/samba-server/complex.rb
index 9021df5..fc3c77d 100644
--- a/src/include/samba-server/complex.rb
+++ b/src/include/samba-server/complex.rb
@@ -114,18 +114,13 @@ module Yast
 
     # Saves service status (start mode and starts/stops the service)
     #
-    # @note For AutoYaST and for command line actions, it uses the old way for
-    # backward compatibility, see {SambaServer#Write}. When the service is
-    # configured using the UI, it is directly saved. See
-    # {Yast2::SystemService#save}.
-    #
     # @param switch_to_reload [Boolean] indicates if restart action must be
     #   replaced with reload. See the Bugzilla #120080 stated in #WriteDialog
     #   comments
     #
     # @return [Boolean] true if service is saved successfully; false otherwise
     def save_status(switch_to_reload)
-      return false if !SambaServer.Write(false)
+      return false unless SambaServer.Write(false)
       services.reload if services.action == :restart && switch_to_reload
       services.save
     end
diff --git a/test/dialog_complex_test.rb b/test/dialog_complex_test.rb
index a0ade2f..41b5430 100644
--- a/test/dialog_complex_test.rb
+++ b/test/dialog_complex_test.rb
@@ -123,64 +123,86 @@ describe "SambaServerComplexInclude" do
     end
 
     context "when running in UI mode" do
-      it "saves the system service" do
-        expect(services).to receive(:save)
+      context "and fails written configuration" do
+        before do
+          allow(Yast::SambaServer).to receive(:Write).and_return(false)
 
-        samba.WriteDialog
-      end
+          it "does not save the system service" do
+            expect(services).to_not receive(:save)
 
-      it "returns :next when sevice is saved successfully" do
-        expect(samba.WriteDialog).to eq(:next)
+            samba.WriteDialog
+          end
+
+          it "returns :abort" do
+            expect(samba.WriteDialog).to eq(:abort)
+          end
+        end
       end
 
-      it "returns :abort when fails saving the service" do
-        allow(services).to receive(:save).and_return(false)
+      context "and configuration is written" do
+        before do
+          allow(Yast::SambaServer).to receive(:Write).and_return(true)
+        end
 
-        expect(samba.WriteDialog).to eq(:abort)
-      end
+        it "saves the system service" do
+          expect(services).to receive(:save)
 
-      context "and action is :restart" do
-        let(:action) { :restart }
+          samba.WriteDialog
+        end
 
-        context "but the service is already running" do
-          let(:service_running) { true }
-          let(:service_on_boot) { true }
+        it "returns :next when sevice is saved successfully" do
+          expect(samba.WriteDialog).to eq(:next)
+        end
 
-          context "with connected users" do
-            it "changes action to :reload" do
-              expect(services).to receive(:reload)
+        it "returns :abort when fails saving the service" do
+          allow(services).to receive(:save).and_return(false)
 
-              samba.WriteDialog
-            end
+          expect(samba.WriteDialog).to eq(:abort)
+        end
 
-            it "reports a message" do
-              allow(services).to receive(:reload)
-              expect(Yast::Report).to receive(:Message)
+        context "and action is :restart" do
+          let(:action) { :restart }
 
-              samba.WriteDialog
+          context "but the service is already running" do
+            let(:service_running) { true }
+            let(:service_on_boot) { true }
+
+            context "with connected users" do
+              it "changes action to :reload" do
+                expect(services).to receive(:reload)
+
+                samba.WriteDialog
+              end
+
+              it "reports a message" do
+                allow(services).to receive(:reload)
+                expect(Yast::Report).to receive(:Message)
+
+                samba.WriteDialog
+              end
             end
-          end
 
-          context "without connected users" do
-            let(:connected_users) { [] }
+            context "without connected users" do
+              let(:connected_users) { [] }
 
-            it "performs the requested action" do
-              expect(services).to_not receive(:reload)
+              it "performs the requested action" do
+                expect(services).to_not receive(:reload)
 
-              samba.WriteDialog
+                samba.WriteDialog
+              end
             end
           end
         end
-      end
 
-      context "and action is NOT :restart" do
-        let (:action) { :reload }
+        context "and action is NOT :restart" do
+          let (:action) { :reload }
 
-        context "and there are connected users" do
-          it "does not report message" do
-            expect(Yast::Report).to_not receive(:Message)
+          context "and there are connected users" do
+            it "does not report message" do
+              expect(Yast::Report).to_not receive(:Message)
 
-            samba.WriteDialog
+              samba.WriteDialog
+            end
           end
         end
       end
```

Despite it seems quite long, note that I simply added - alongside minor changes - two new `contexts` nested in "when running in UI mode"

* "and fails written configuration"
* "and configuration is written" 

and two new examples in the first one.

I hope you find it useful.

Don't hesitate to reach me again if I can continue to help you.

File: src/include/samba-server/complex.rb [hunk 1]
```diff
@@ -125,12 +125,9 @@ def WriteDialog
     #
     # @return [Boolean] true if service is saved successfully; false otherwise
     def save_status(switch_to_reload)
-      if Mode.auto || Mode.commandline
-        SambaServer.Write(false)
-      else
-        services.reload if services.action == :restart && switch_to_reload
-        services.save
-      end
+      return false if !SambaServer.Write(false)
```
Comment (dgdavid) [inline src/include/samba-server/complex.rb]: Better use `unless` instead `if !`

**Comment by kernevil** [comment]
Thanks @dgdavid, all issues resolved.

**Pull request merged**